web_src/js/features/repo-view-file-tree-sidebar.ts

```diff
@@ -0,0 +1,103 @@
+import {createApp, ref} from 'vue';
+import {toggleElem} from '../utils/dom.ts';
+import {GET, PUT} from '../modules/fetch.ts';
+import {registerGlobalInitFunc} from '../modules/observer.ts';
+import ViewFileTree from '../components/ViewFileTree.vue';
+
+async function toggleSidebar(sidebarEl: HTMLElement, visibility: boolean) {
+  const showBtnEl = sidebarEl.parentElement.querySelector('.show-tree-sidebar-button');
+  const containerClassList = sidebarEl.parentElement.classList;
+  containerClassList.toggle('repo-grid-tree-sidebar', visibility);
+  containerClassList.toggle('repo-grid-filelist-only', !visibility);
+  toggleElem(sidebarEl, visibility);
+  toggleElem(showBtnEl, !visibility);
+
+  // FIXME: need to remove "full height" style from parent element
+
+  if (!sidebarEl.hasAttribute('data-is-signed')) return;
+
+  // save to session
+  await PUT('/repo/preferences', {
+    data: {
+      show_file_view_tree_sidebar: visibility,
+    },
+  });
+}
+
+function childrenLoader(sidebarEl: HTMLElement) {
+  return async (path: string, recursive?: boolean) => {
+    const fileTree = sidebarEl.querySelector('#view-file-tree');
+    const apiBaseUrl = fileTree.getAttribute('data-api-base-url');
+    const refTypeNameSubURL = fileTree.getAttribute('data-current-ref-type-name-sub-url');
+    const response = await GET(encodeURI(`${apiBaseUrl}/tree/${refTypeNameSubURL}/${path ?? ''}?recursive=${recursive ?? false}`));
```

Comment:
I do not think it is right. You need to encode the "parts" correctly.

For example: `encodeURI` doesn't encode `?`, `#`, `&` etc. What if the path contains them?

Reply:
As far as I know, I can only use encodeURIComponent to handle each part of the path. I'll try to make the changes.

Reply:
See `pathEscapeSegments` in TS